http/client.go

```diff
@@ -77,18 +65,25 @@ func NewClient() Clienter {
 			Transport: DefaultTransport,
 		},
 	}
-	return newClient
 }
 
 // NewClientWithAwsSigner return a new client with aws signer profile.
 func NewClientWithAwsSigner(awsFilename, awsProfile, awsRegion, awsService string) (Clienter, error) {
-	newClient := *DefaultClient
+	newClient := &Client{
```

Comment:
Then newClient is equal to DefultClient and `return &newClient`?

Reply:
its all the same right ? i am taking in a client and return pointer to struct which is the clienter interface